### PR TITLE
lossess compress assets

### PR DESCRIPTION
**Pull request by TheDen**
Resolves: Very (natural) minor improvement in asset sizes (w/ lossless compression)

PR is a run of [imageoptim](https://imageoptim.com/mac) on the `assets` folder, (with settings shown below).

Minification is completely lossless, and in total saves 810KBs of data.

image

image

Apologies if this PR may be too trivial, but it's basically a "free" reduction in overhead I immediately noticed when going through the codebase.

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [x] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [x] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

**Comment by Jojo-Schmitz** [comment]
typoe: 'lossess' vs. 'loss**l**ess'

File: assets/msc-icon-master.svg [hunk 1]
```diff
@@ -1,57 +1 @@
...
... mode="normal" in2="shape" result="effect1_innerShadow"/></filter><filter id="filter2_d" x="272.869" y="35.463" width="165.331" height="162.684" filterUnits="userSpaceOnUse" color-interpolation-filters="sRGB"><feFlood flood-opacity="0" result="BackgroundImageFix"/><feColorMatrix in="SourceAlpha" type="matrix" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"/><feOffset dy="13"/><feGaussianBlur stdDeviation="12"/><feColorMatrix type="matrix" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.337409 0"/><feBlend mode="normal" in2="BackgroundImageFix" result="effect1_dropShadow"/><feBlend mode="normal" in="SourceGraphic" in2="effect1_dropShadow" result="shape"/></filter><linearGradient id="paint0_linear" x1="432" y1="-87.344" x2="127.878" y2="214.761" gradientUnits="userSpaceOnUse"><stop stop-color="#BBB"/><stop offset=".293" stop-color="#D9D9D9"/><stop offset=".637" stop-color="#F5F5F5"/><stop offset="1" stop-color="#fff"/></linearGradient><linearGradient id="paint1_linear" x1="342.965" y1="89.191" x2="319.17" y2="124.162" gradientUnits="userSpaceOnUse"><stop stop-color="#DEDEDE"/><stop offset=".446" stop-color="#EEE"/><stop offset="1" stop-color="#F8F8F8"/></linearGradient></defs></svg>
```
Comment (Jojo-Schmitz) [inline assets/msc-icon-master.svg:1]: Loosing linefeeds in a text format lile SVG is not good

File: assets/mscx-icon.svg [hunk 2]
```diff
@@ -1,57 +1 @@
...
... mode="normal" in2="shape" result="effect1_innerShadow"/></filter><filter id="filter2_d" x="272.869" y="35.463" width="165.331" height="162.684" filterUnits="userSpaceOnUse" color-interpolation-filters="sRGB"><feFlood flood-opacity="0" result="BackgroundImageFix"/><feColorMatrix in="SourceAlpha" type="matrix" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"/><feOffset dy="13"/><feGaussianBlur stdDeviation="12"/><feColorMatrix type="matrix" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.337409 0"/><feBlend mode="normal" in2="BackgroundImageFix" result="effect1_dropShadow"/><feBlend mode="normal" in="SourceGraphic" in2="effect1_dropShadow" result="shape"/></filter><linearGradient id="paint0_linear" x1="432" y1="-87.344" x2="127.878" y2="214.761" gradientUnits="userSpaceOnUse"><stop stop-color="#BBB"/><stop offset=".293" stop-color="#D9D9D9"/><stop offset=".637" stop-color="#F5F5F5"/><stop offset="1" stop-color="#fff"/></linearGradient><linearGradient id="paint1_linear" x1="342.965" y1="89.191" x2="319.17" y2="124.162" gradientUnits="userSpaceOnUse"><stop stop-color="#DEDEDE"/><stop offset=".446" stop-color="#EEE"/><stop offset="1" stop-color="#F8F8F8"/></linearGradient></defs></svg>
```
Comment (Jojo-Schmitz) [inline assets/mscx-icon.svg:1]: same

File: assets/mscz-icon.svg [hunk 3]
```diff
@@ -1,57 +1 @@
...
... mode="normal" in2="shape" result="effect1_innerShadow"/></filter><filter id="filter2_d" x="272.869" y="35.463" width="165.331" height="162.684" filterUnits="userSpaceOnUse" color-interpolation-filters="sRGB"><feFlood flood-opacity="0" result="BackgroundImageFix"/><feColorMatrix in="SourceAlpha" type="matrix" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"/><feOffset dy="13"/><feGaussianBlur stdDeviation="12"/><feColorMatrix type="matrix" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.337409 0"/><feBlend mode="normal" in2="BackgroundImageFix" result="effect1_dropShadow"/><feBlend mode="normal" in="SourceGraphic" in2="effect1_dropShadow" result="shape"/></filter><linearGradient id="paint0_linear" x1="432" y1="-87.344" x2="127.878" y2="214.761" gradientUnits="userSpaceOnUse"><stop stop-color="#BBB"/><stop offset=".293" stop-color="#D9D9D9"/><stop offset=".637" stop-color="#F5F5F5"/><stop offset="1" stop-color="#fff"/></linearGradient><linearGradient id="paint1_linear" x1="342.965" y1="89.191" x2="319.17" y2="124.162" gradientUnits="userSpaceOnUse"><stop stop-color="#DEDEDE"/><stop offset=".446" stop-color="#EEE"/><stop offset="1" stop-color="#F8F8F8"/></linearGradient></defs></svg>
```
Comment (Jojo-Schmitz) [inline assets/mscz-icon.svg:1]: same

File: assets/musescore-icon-round.svg [hunk 4]
```diff
@@ -1,12 +1 @@
...
...2 229.23 512 512 512z" fill="url(#a)" fill-rule="evenodd"/><g fill="#fff"><path d="m402.927 450.435c-20.773-28.727-61.722-47.584-99.879-47.584-63.048 0-121.68 38.732-121.68 145.021v178.196c0 12.077 9.797 21.801 21.874 21.801 12.085 0 21.881-9.797 21.881-21.801v-178.196c-.022-80.778 38.959-101.341 77.925-101.341s77.933 32.391 77.933 87.057v192.48c0 12.077 9.796 21.801 21.873 21.801h.095.095c12.078 0 21.874-9.724 21.874-21.801v-192.48c0-54.666 38.966-87.057 77.932-87.057 24.738 0 77.925 10.987 77.925 87.057l1.48 64.647c0 119.452 80.177 149.634 131.199 149.634s127.453-35.242 129.012-140.93v-169.122c0-29.092-43.878-29.092-43.878 0v169.122c0 80.177-59.623 97.255-85.134 97.255s-87.466-12.018-87.466-105.959l-1.458-61.109c0-90.605-56.853-137.919-121.68-137.919-38.157 0-79.149 22.501-99.923 51.228z"/><path d="m621.197 340.896c0-43.733 32.209-91.11 87.903-91.11s87.795 47.377 87.795 91.11c0 32.8 43.835 32.8 43.835 0 0-76.533-57.713-134.843-131.63-134.843-73.916 0-131.614 58.31-131.614 134.843.007 32.8 43.711 32.8 43.711 0z"/><path d="m742.185 342.117c0-15.949-12.916-28.864-28.864-28.864s-28.864 12.915-28.864 28.864c0 15.948 12.916 28.863 28.864 28.863s28.864-12.915 28.864-28.863z"/></g></svg>
```
Comment (Jojo-Schmitz) [inline assets/musescore-icon-round.svg:1]: same

File: assets/musicxml-icon-master.svg [hunk 5]
```diff
@@ -1,64 +1 @@
...
...8 C45.273,168.555 49.529,162.492 55.59,159.809 L61.263,204.02 L62.588,214.278 C63.452,219.151 64.307,224.003 65.125,228.798 C66.125,235.286 66.477,240.91 66.166,245.513 C65.747,251.888 63.758,257.097 60.247,261.006 C58.013,263.491 55.403,265.061 52.452,265.931 C56.413,261.969 58.866,256.501 58.866,250.453 C58.866,238.378 49.074,228.581 36.996,228.581 C33.643,228.581 30.505,229.393 27.667,230.74 C27.307,230.895 26.968,231.128 26.613,231.314 C25.834,231.739 25.054,232.164 24.327,232.676 C24.078,232.863 23.84,233.079 23.595,233.281 C18.707,237.097 15.504,242.897 15.22,249.51 C15.22,249.52 15.209,249.531 15.209,249.541 C15.209,249.546 15.209,249.551 15.209,249.557 C15.196,249.868 15.119,250.147 15.119,250.453 C15.119,250.712 15.186,250.945 15.197,251.193 C15.186,258.277 18.13,264.372 24.036,269.311 C29.62,273.977 36.313,276.591 43.928,277.088 C48.255,277.368 52.433,276.871 56.335,275.603 C59.524,274.562 62.61,272.983 65.484,270.916 C72.444,265.894 76.234,259.332 76.756,251.384 C77.155,245.321 76.85,238.149 75.819,229.973 L72.981,211.544 L65.825,158.645 C73.809,160.177 79.852,167.188 79.852,175.613 C79.852,181.599 77.915,191.044 74.301,195.989 L76.55,210.161 Z" id="Path"/></g></g></svg>
```
Comment (Jojo-Schmitz) [inline assets/musicxml-icon-master.svg:1]: same

File: assets/mxl-icon.svg [hunk 6]
```diff
@@ -1,64 +1 @@
...
....895-6.322-9.264-14.98-14.317-25.745-15.026-1.232-.081-2.543-.075-3.895.025l-2.304-17.191-9.668 7.112 1.483 12.194c-5.789 2.393-10.827 6.17-15.017 11.255-4.823 5.924-7.508 12.443-7.964 19.382-.466 7.208 1.142 13.81 4.782 19.583 1.895 3.081 4.507 5.82 7.498 8.058 4.906 3.65 10.563 3.376 11.459 1.393.906-1.983-2.455-5.095-5.09-9.248-1.502-2.351-2.242-5.173-2.242-8.497 0-7.053 4.256-13.116 10.317-15.799l5.673 44.211 1.325 10.258c.864 4.873 1.719 9.725 2.537 14.52 1 6.488 1.352 12.112 1.041 16.715-.419 6.375-2.408 11.584-5.919 15.493-2.234 2.485-4.844 4.055-7.795 4.925 3.961-3.962 6.414-9.43 6.414-15.478 0-12.075-9.792-21.872-21.87-21.872-3.353 0-6.491.812-9.329 2.159-.36.155-.699.388-1.054.574-.779.425-1.559.85-2.286 1.362-.249.187-.487.403-.732.605-4.888 3.816-8.091 9.616-8.375 16.229 0 .01-.011.021-.011.031v.016c-.013.311-.09.59-.09.896 0 .259.067.492.078.74-.011 7.084 2.933 13.179 8.839 18.118 5.584 4.666 12.277 7.28 19.892 7.777 4.327.28 8.505-.217 12.407-1.485 3.189-1.041 6.275-2.62 9.149-4.687 6.96-5.022 10.75-11.584 11.272-19.532.399-6.063.094-13.235-.937-21.411l-2.838-18.429-7.156-52.899c7.984 1.532 14.027 8.543 14.027 16.968 0 5.986-1.937 15.431-5.551 20.376z"/></g></g></svg>
```
Comment (Jojo-Schmitz) [inline assets/mxl-icon.svg:1]: same

File: assets/xml-icon.svg [hunk 7]
```diff
@@ -1,60 +1 @@
...
...8-.02725.98805-.13631.34356-.1159.63316-.28001.88352-.4987.25036-.21868.42347-.45768.5051-.71801.14098-.29305.22577-.70843.26053-1.2309.02348-.35322.01146-.70898-.02909-1.0668-.04103-.35044-.10972-.82116-.20666-1.4048-.09744-.57629-.16763-1.0249-.20446-1.3307-.43406.07984-.87892.10715-1.3412.0741-.7752-.05546-1.497-.26979-2.1648-.6503-.66783-.38051-1.242-.88002-1.7291-1.5064-.48001-.62585-.84134-1.3171-1.083-2.0886-.23503-.76359-.33249-1.5543-.28576-2.3643.07842-.74855.26148-1.46.56255-2.1187.30156-.66604.67178-1.2902 1.1173-1.8646s.90183-1.0963 1.3685-1.5583c.47325-.45412 1.0909-1.0384 1.8668-1.7444z" fill="#4d4d4d" stroke="#333" stroke-width=".75"/></g><g opacity=".4"><path d="m15.209 40.916h1.6l1.16-1.9 1.16 1.9h1.6l-1.97-2.86 1.69-2.68h-1.5l-.98 1.67-1-1.67h-1.5l1.7 2.68z"/><path d="m21.344 40.916h1.33v-2.73c0-.55.05-.92.14-1.15.16-.36.5-.57.94-.57.36 0 .67.13.85.35.16.21.23.54.23 1.06v3.04h1.33v-2.73c0-.62.05-.93.19-1.19.17-.34.51-.53.93-.53.33 0 .61.12.79.34.18.21.25.52.25 1.07v3.04h1.33v-3.2c0-.67-.12-1.21-.36-1.59-.35-.56-1.05-.88-1.88-.88-.78 0-1.33.26-1.78.86-.38-.6-.87-.86-1.63-.86-.65 0-1.05.18-1.44.65v-.52h-1.22z"/><path d="m30.739 40.916h1.33v-7.39h-1.33z"/></g></svg>
```
Comment (Jojo-Schmitz) [inline assets/xml-icon.svg:1]: same

**Comment by TheDen** [comment]
Sorry about the commit message, will fix it (either a rewrite history or with a new commit that we can squash).

On losing linefeeds in a text format SVG not being good, I didn't know about that. Do you have an example of some cases where it would break things?

Regardless, I could preserve the linefeeds in a new commit.

**Comment by Jojo-Schmitz** [comment]
If you want to 'hand edit' such a file, you may hit line-length limits in it the editor

**Comment by RobFog** [comment]
Image compression has been suggested previously but was not done due to objections of this kind: https://github.com/musescore/MuseScore/pull/5688#issuecomment-584522668

**Comment by shoogle** [comment]
Unfortunately there's little point in compressing files *after* they have already been added to the repository. Git tracks the difference between successive versions of a file, so any change made to a file will always cause the repository to grow larger, even if you made the file itself smaller.

A better solution would be to apply the compression at compile time, but considering the program is already 130+ MB it just isn't worth it to save 1 or 2 MB. Next time we update the assets we should compress them *before* we add them to the repository.

**Pull request closed without merging**